src/mlpack/methods/lsh/objectivefunction.hpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file objectivefunction.hpp
+ * @author Yannis Mentekidis
+ *
+ * This file implements a class that describes an objective function for
+ * minimization. It is used by the LSH model to fit a curve of the form
+ * E(k, N) = \alpha \cdot k ^ \beta \cdot N^\gamma
+ * to a certain statistic E, which can be either the arithmetic or the geometric
+ * mean of distances of a random point and its k-Nearest Neighbors.
+ *
+ * The objective function to minimize is the mean squared error (MSE):
+ *
+ * Error =\sum_{i=0}^{M} (y(i) - \alpha \cdot k ^ \beta \cdot N^\gamma)^2
+ *
+ * The class is designed for use with the L_BFGS optimizer, which is what the
+ * lshmodel class uses.
+ */
+
+#ifndef MLPACK_METHODS_NEIGHBOR_SEARCH_LSH_DEFAULT_OBJECTIVE_FUNCTION_HPP
+#define MLPACK_METHODS_NEIGHBOR_SEARCH_LSH_DEFAULT_OBJECTIVE_FUNCTION_HPP
+
+namespace mlpack {
+namespace neighbor {
+
+class DefaultObjectiveFunction
+{
+  public:
+    //! Default constructor - do nothing.
+    DefaultObjectiveFunction() { };
+
+    /**
+     * Parameterized constructor.
+     *
+     * @param xData Vector of x - the sizes of the reference set when performing
+     *    kNN.
+     * @param kData Vector of k - the kth nearest neighbor for which we
+     *    calculated the statistic.
+     * @param yData Matrix of y, one for each (x, k) value.
+     */
+    DefaultObjectiveFunction(const arma::Col<size_t>& xData, 
+                             const arma::Col<size_t>& kData, 
+                             const arma::mat& yData)
+      : xData(&xData), kData(&kData), yData(&yData)  { };
+
+    //! Return the number of functions
```

Comment:
If you are only using L-BFGS, you should be able to remove this function. It should only be needed by SGD-like optimizers.